### PR TITLE
fix(perf): correct metrics snapshot selection and improve peak tracking

### LINKED ISSUE #3943: Improve performance metrics PR message report
### Describe the bug

The performance metrics do not show peakCpuUsage correctly also some fields are in wrong order which is making it hard to use

### Describe the expected behavior

Showing correct metrics for all types of memory usage with correct snapshots and correct order of parameters

### To Reproduce

check logs from performance tests

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by boris-bonin**
## Description

- Select max-memory snapshot as representative instead of max-CPU (startup spikes at 5s no longer win over steady-state workload)
- Add peakCpuInMillicores and peak snapshot names to summary JSON
- Track running peakMemoryInMebibytes in each individual snapshot
- Fix cleanup loop extension mismatch (Set.has compared without .json)
- Copy summary JSON to metrics artifact directory in workflow
- Guard for missing postgresPodName in transaction count lookup
- Upgrade transaction count error logging from debug to warn

### Related Issues

* Closes #3943
* Related to #

### Pull request (PR) checklist

* \[ ] This PR added tests (unit, integration, and/or end-to-end)
* \[ ] This PR updated documentation
* \[ ] This PR added no TODOs or commented out code
* \[ ] This PR has no breaking changes
* \[ ] Any technical debt has been documented as a separate issue and linked to this PR
* \[ ] Any `package.json` changes have been explained to and approved by a repository manager
* \[ ] All related issues have been linked to this PR
* \[ ] All changes in this PR are included in the description
* \[ ] When this PR merges the commits will be squashed and the title will be used as the commit message, the 'commit message guidelines' below have been followed

### Testing

* \[ ] This PR added unit tests
* \[ ] This PR added integration/end-to-end tests
* \[ ] These changes required manual testing that is documented below
* \[ ] Anything not tested is documented

The following manual testing was done:

* TBD

The following was not tested:

* TBD

<details>
<summary>
Commit message guidelines
</summary>
We use 'Conventional Commits' to ensure that our commit messages are easy to read, follow a consistent format, and for automated release note generation. Please follow the guidelines below when writing your commit messages:

1. BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.  NOTE: currently breaking changes will only bump the MAJOR version.
2. The title is prefixed with one of the following:

| Prefix    | Description                                         | Semantic Version Update | Captured in Release Notes |
|-----------|-----------------------------------------------------|-------------------------|---------------------------|
| feat:     | a new feature                                       | MINOR                   | Yes                       |
| fix:      | a bug fix                                           | PATCH                   | Yes                       |
| perf:     | performance                                         | PATCH                   | Yes                       |
| refactor: | code change that isn't feature or fix               | none                    | No                        |
| test:     | adding missing tests                                | none                    | No                        |
| docs:     | changes to documentation                            | none                    | Yes                        |
| build:    | changes to build process                            | none                    | No                        |
| ci:       | changes to CI configuration                         | none                    | No                        |
| style:    | formatting, missing semi-colons, etc                | none                    | No                        |
| chore:    | updating grunt tasks etc; no production code change | none                    | No                        |

</details>

### CODE REVIEW

File: test/e2e/commands/performance.test.ts
Comment (jan-milenkov) [inline test/e2e/commands/performance.test.ts:1]: Improve types, use `type` and `interfaces` if needed, avoid duplicating types and using `as` type casting

**Review by jan-milenkov — CHANGES_REQUESTED** [review]

File: test/e2e/commands/performance.test.ts [hunk 1]
```diff
@@ -264,6 +280,18 @@ export async function logMetrics(startTime: Date): Promise<void> {
     undefined,
     events,
   );
+
+  // Track running peak memory and inject it into the snapshot file
+  const snapshotPath: string = PathEx.join(tartgetDirectory, `${elapsedMilliseconds}.json`);
```
Comment (jan-milenkov) [inline test/e2e/commands/performance.test.ts:285]: is there a reason for the `elapsedMilliseconds` to be the filename?

**Review by jan-milenkov — APPROVED** [review]
Looks good to me, please note the question in the comment

File: test/e2e/commands/performance.test.ts (hunk 1 above)
Comment (jan-milenkov) [reply test/e2e/commands/performance.test.ts:285]: @boris-bonin

Comment (boris-bonin) [reply test/e2e/commands/performance.test.ts:285]: It serves as a unique timestamp-based identifier for each snapshot. Each call to logMetrics produces a file named by how many milliseconds have elapsed since startTime. This gives chronological ordering and uniqueness.

**Review by jan-milenkov — APPROVED** [review]

**Review by jeromy-cannon — APPROVED** [review]
